lib/editor/components/timetable/TripSeriesModal.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+// $FlowFixMe: Flow doesn't know about useState: https://stackoverflow.com/questions/53105954/cannot-import-usestate-because-there-is-no-usestate-export-in-react-flow-with
+import React, {useState} from 'react'
+import {Button, Modal} from 'react-bootstrap'
+
+import * as tripActions from '../../actions/trip'
+import {getComponentMessages} from '../../../common/util/config'
+import HourMinuteInput from '../HourMinuteInput'
+import type {Trip} from '../../../types'
+
+type Props = {
+  addNewTrip: typeof tripActions.addNewTrip,
+  constructNewRow: (trip: ?Trip, tripSeriesStartTime: number) => ?Trip,
+  onClose: () => void,
+  show: boolean
+}
+
+const TripSeriesModal = (props: Props) => {
+  const [startTime, setStartTime] = useState(null)
+  const [headway, setHeadway] = useState(null)
+  const [endTime, setEndTime] = useState(null)
+
+  const messages = getComponentMessages('TripSeriesModal')
+
+  const onClickGenerate = () => {
```

Comment:
Consider using `useCallback` (so that this function doesn't get recreated each render).

Reply:
Sure, I used `useCallback` in https://github.com/ibi-group/datatools-ui/pull/892/commits/69ed91b487e9585118c726639dd711c98d0c2df1